### PR TITLE
Add processing parameter types for database schema and table name

### PR DESCRIPTION
**Pull request by nyalldawson**
Replaces the previous ad-hoc Python postgis widget wrappers with proper first class c++ dedicated parameters for schemas and tables, with wrappers built off the core connections api. 

Refs NRCan Contract#3000707093

### CODE REVIEW

**Comment by nyalldawson** [comment]
Ping @alexbruy

File: src/core/processing/qgsprocessingparameters.cpp [hunk 1]
```diff
@@ -6624,3 +6658,296 @@ QgsProcessingParameterProviderConnection *QgsProcessingParameterProviderConnecti
...
+
+void QgsProcessingParameterDatabaseSchema::setParentConnectionParameterName( const QString &name )
+{
+  mParentConnectionParameterName = name;
+}
+
+QVariantMap QgsProcessingParameterDatabaseSchema::toVariantMap() const
+{
+  QVariantMap map = QgsProcessingParameterDefinition::toVariantMap();
+  map.insert( QStringLiteral( "mParentLayerParameterName" ), mParentConnectionParameterName );
```
Comment (alexbruy) [inline src/core/processing/qgsprocessingparameters.cpp]: Shouldn't `mParentLayerParameterName` be a `mParentConnectionParameterName`?

File: src/core/processing/qgsprocessingparameters.cpp [hunk 2]
```diff
@@ -6624,3 +6658,296 @@ QgsProcessingParameterProviderConnection *QgsProcessingParameterProviderConnecti
...
+  return map;
+}
+
+bool QgsProcessingParameterDatabaseSchema::fromVariantMap( const QVariantMap &map )
+{
+  QgsProcessingParameterDefinition::fromVariantMap( map );
+  mParentConnectionParameterName = map.value( QStringLiteral( "mParentLayerParameterName" ) ).toString();
```
Comment (alexbruy) [inline src/core/processing/qgsprocessingparameters.cpp]: Same as above?

**Review by alexbruy — APPROVED** [review]

File: src/core/processing/qgsprocessingparameters.cpp (hunk 2 above)
Comment (nyalldawson) [reply src/core/processing/qgsprocessingparameters.cpp]: Fixed

**Comment by nyalldawson** [comment]
UNrelated test failure

**Pull request merged**